sympy/sets/tests/test_fancysets.py

```diff
@@ -441,6 +441,18 @@ def test_imageset_intersect_real():
     assert s.intersect(S.Reals) == imageset(Lambda(n, 2*n*pi - pi/4), S.Integers)
 
 
+def test_imageset_intersect_interval():
```

Comment:
I needed help with the case of complex functions.
Since non-real `Interval` cannot be supported, this method won't work.